src/PublicSale/Allowlist.mo

```diff
@@ -0,0 +1,31 @@
+import HashMap "mo:base/HashMap";
+
+import AccountIdentifier "mo:principal/AccountIdentifier";
+
+import Types "types";
+
+module {
+    public func isInAllowlist(
+        caller    : Principal,
+        allowlist : HashMap.HashMap<Types.AccountIdentifier, Nat8>
+    ) : ?Types.AccountIdentifier {
+        let accountId = AccountIdentifier.fromPrincipal(caller, null); // 0-subaccount
```

Comment:
Ah, you meant can we assume subaccount 0 *here*. 🤔 This may break, because I have no idea what accounts people sent me. Worse case scenario, the pre sale lasts several days which gives us plenty of time to provide support if it does break.

Not sure how to handle this long term. One option would be to only support principals on the allow list. That might be simplest / best. Not sure if there's a better way. We can't really get principals from people for this round though.